### PR TITLE
Tidy 6.6 Framework release notes

### LINKED ISSUE #35150: Tidy 6.6 `framework.rst`

### PR DESCRIPTION
**Pull request by jhaigh0**
**Description of work.**

Tidied the 6.6 Framework release notes file.

**This must be merged after #35137 !!**

**To test:**

 - Build `docs-html`
 - Open `framework.html`
 - Read through the notes and check that the grammar and spelling are good
 - Check that links work and go where you expect

Fixes #35150

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v6.6.0/framework.rst [hunk 1]
```diff
@@ -7,28 +7,33 @@ Framework Changes
 
 New Features
 ------------
-- the approach for inverting large tridiagonal matrices has been improved so that the analytical method can still be used even for large matrices, resulting in a performance improvement. This type of matrix inversion is used when calculating the interpolation errors when performing spline interpolation
+- The approach for inverting large tridiagonal matrices has been improved so that the analytical method can still be used even for large matrices, resulting in a performance improvement. This type of matrix inversion is used when calculating the interpolation errors when performing spline interpolation.
```
Comment (rbauststfc) [inline docs/source/release/v6.6.0/framework.rst:10]: Just a thought, but would sentence:

> This type of matrix inversion is used when calculating the interpolation errors when performing spline interpolation.

Read better as:

> This type of matrix inversion is used when calculating the interpolation errors while performing spline interpolation.

**Review by rbauststfc — COMMENTED** [review]
This looks good and all the links are working as expected. I've added a small optional suggestion, but feel free to ignore if you don't agree.

Finally, I just wanted to confirm that we normally keep empty sections in the file? For example, there aren't any changes listed under Fit Functions, Data Objects New Features or Python Bugfixes for this release, but would we usually still keep these in?

File: docs/source/release/v6.6.0/framework.rst (hunk 1 above)
Comment (jhaigh0) [reply docs/source/release/v6.6.0/framework.rst:10]: Yes, I like that change, thank you for suggesting.

The headings stay there until we are sure no more release notes will be added; so removing them is one of the last things to do before release.

**Review by rbauststfc — APPROVED** [review]
This is all looking good and thanks for explaining about the empty headings, that makes sense 👍

**Pull request merged**